docs/proposals/user-namespace.md

```diff
@@ -0,0 +1,206 @@
+some not.  Any runtime behavior should be isolated to the kubelet
+and the container runtime.
+
+### Detecting and Driving Private User Namespaces Behavior
+
+In order to Do The Right Thing in any solution the system must be able 
+to detect that private user namespaces are in use.  
+
+In Docker, this is a daemon setting and is [exposed via 
+the info endpoint](https://github.com/docker/docker/commit/ae74092e450f1f2665b90257b65513cc0c19702f).
+This information is available in Docker 1.13+ only.
+
+However, in rkt this is a run parameter so it is
+unknown if private user namespaces will be used until the container launch
+command is created.  
+
+As a consistent and backwards compatible solution to both driving any behavior to escape 
+user namespaces (see system defaulting below) and to also control
+the use of `--private-users` in the rkt `run` command a feature
+gate can be added to the Kubelet as was done in the [experimental
+support](https://github.com/kubernetes/kubernetes/pull/31169/files#diff-c5e4440d8576a91980ab36fe63556326).
+
+### Volumes
+
+The system must ensure that the remapped UID and GID have
+access to the correct directories under `/var/lib/kubelet/pods` in order
+to perform mounting.  Currently the Kubelet root directory is created
+with 0750 permissions and are owned by root:root.  Since pods are running
+with at least the root group (`runAsUser` may be specified as well as 
+supplemental groups) the pod has access to read these directories.  In
+a remap environment the group itself may change so the pod will no longer
+be able to access these directories.  In order to fix this the root 
+directory should be given 0755 access in the Kubelet's `setupDataDirs`.
+
+Beneath the Kubelet root directory each pod is given it's own directory
+named by a unique identifier containing:
+
+* containers
+* plugins
+* volumes
+
+Rather than increasing the access on directories under the root directory,
+the pod directory (and sub directories) may be owned by the remapped
+GID.  This is very similar to the `fsGroup`, however `fsGroup` is not applied
+to items like the termination logs.
+
+How this GID is retrieved should be a container runtime detail.  This
+leaves the container runtimes free to manage the permissions of the 
+directories under the pod directory (available via injecting the
+`RuntimeHelper` into the runtime implementation) in a way that suits
+their ability to provide the GID.  For Docker this is the value in
+`/etc/subgid`.
+
+There is movement towards being able to [specify](https://github.com/docker/docker/issues/28593) 
+or [lock ranges](https://github.com/coreos/rkt/issues/1090)
+of UIDs/GIDs in the runtime community.  As this work progresses it 
+makes sense that the ability to allocate a range should belong to the
+orchestration system.  This would allow Kuberenetes to know exactly
+what GID requires access to the directories and ensure the correct
+access is granted.  
```

Comment:
Does this imply some future integration between the runtime and PSP allocators for these IDs?